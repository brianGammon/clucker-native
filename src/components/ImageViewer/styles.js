/* @flow */
import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  modalContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.75)',
  },
  closeButton: {
    position: 'absolute',
    right: 10,
    top: 10,
  },
  closeIcon: {
    fontSize: 60,
  },
});
