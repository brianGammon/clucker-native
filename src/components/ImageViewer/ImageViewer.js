/* @flow */
import React from 'react';
import { View, Icon, Button } from 'native-base';
import { Dimensions, Modal, Image } from 'react-native';
import styles from './styles';

type Props = {
  url: string,
  showModal: boolean,
  toggleModal: (visible: boolean) => void,
};

const ImageViewer = ({ url, showModal, toggleModal }: Props) => {
  let { height, width } = Dimensions.get('window');
  return (
    <Modal
      animationType="fade"
      transparent
      visible={showModal}
      supportedOrientations={['portrait', 'landscape']}
      onOrientationChange={() => {
        const { height: newHeight, width: newWidth } = Dimensions.get('window');
        height = newHeight;
        width = newWidth;
      }}
      onRequestClose={() => {}}
    >
      <View style={styles.modalContainer}>
        <Button
          large
          light
          transparent
          style={styles.closeButton}
          onPress={() => {
            toggleModal(!showModal);
          }}
        >
          <Icon style={styles.closeIcon} name="close" />
        </Button>
        <Image
          resizeMode="cover"
          style={{
            width: width < height ? width : height,
            height: width < height ? width : height,
          }}
          source={
            url
              ? { uri: url }
              : require('../../assets/default-profile-photo.png')
          }
        />
      </View>
    </Modal>
  );
};

export default ImageViewer;
