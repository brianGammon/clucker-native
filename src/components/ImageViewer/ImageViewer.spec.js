import React from 'react';
import ImageViewer from '.';

test('ImageViewer renders properly with image URL', () => {
  const wrapper = shallow(
    <ImageViewer
      showModal
      toggleModal={() => {}}
      url="https://example.com/image.jpg"
    />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('ImageViewer renders properly with no image URL provided', () => {
  const wrapper = shallow(<ImageViewer showModal toggleModal={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('ImageViewer renders properly showModal set to false', () => {
  const wrapper = shallow(
    <ImageViewer
      showModal={false}
      toggleModal={() => {}}
      url="https://example.com/image.jpg"
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
