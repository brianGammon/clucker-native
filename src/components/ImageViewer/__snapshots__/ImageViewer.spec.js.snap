// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageViewer renders properly showModal set to false 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onOrientationChange={[Function]}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  transparent={true}
  visible={false}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.75)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "position": "absolute",
          "right": 10,
          "top": 10,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="white"
        name="ios-close"
        size={12}
        style={
          Object {
            "fontSize": 60,
          }
        }
      />
    </TouchableOpacity>
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        Object {
          "height": 750,
          "width": 750,
        }
      }
    />
  </Component>
</Component>
`;

exports[`ImageViewer renders properly with image URL 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onOrientationChange={[Function]}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  transparent={true}
  visible={true}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.75)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "position": "absolute",
          "right": 10,
          "top": 10,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="white"
        name="ios-close"
        size={12}
        style={
          Object {
            "fontSize": 60,
          }
        }
      />
    </TouchableOpacity>
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        Object {
          "height": 750,
          "width": 750,
        }
      }
    />
  </Component>
</Component>
`;

exports[`ImageViewer renders properly with no image URL provided 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onOrientationChange={[Function]}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  transparent={true}
  visible={true}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.75)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "position": "absolute",
          "right": 10,
          "top": 10,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="white"
        name="ios-close"
        size={12}
        style={
          Object {
            "fontSize": 60,
          }
        }
      />
    </TouchableOpacity>
    <Image
      resizeMode="cover"
      source={
        Object {
          "testUri": "../../../src/assets/default-profile-photo.png",
        }
      }
      style={
        Object {
          "height": 750,
          "width": 750,
        }
      }
    />
  </Component>
</Component>
`;
